test/gpu/miopen.cpp

```diff
@@ -3359,5 +3359,50 @@ struct test_logsoftmax_1 : verify_program<test_logsoftmax_1<Axis>>
 
 template struct test_logsoftmax_1<0>;
 template struct test_logsoftmax_1<1>;
+#endif
+struct test_split : verify_program<test_split>
+{
+    migraphx::program create_program() const
+    {
+        migraphx::program p;
+        migraphx::shape s{migraphx::shape::int32_type, {3, 3, 3}};
+        migraphx::shape s2{migraphx::shape::int32_type, {27}};
+        auto x      = p.add_parameter("x", s);
+        auto split0 = p.add_instruction(migraphx::op::split{0, {2, 1}}, x);
+        auto split1 = p.add_instruction(migraphx::op::split{1, {2, 1}}, x);
+        p.add_instruction(migraphx::op::add{}, split1, split0);
+        return p;
+    }
+};
+
+struct test_split_selector : verify_program<test_split_selector>
+{
+    migraphx::program create_program() const
+    {
+        migraphx::program p;
+        migraphx::shape s{migraphx::shape::int32_type, {3, 3, 3}};
+        migraphx::shape s2{migraphx::shape::int32_type, {27}};
+        auto x      = p.add_parameter("x", s);
+        auto split0 = p.add_instruction(migraphx::op::split{0, {1, 1, 1}, {0, 0}}, x);
+        auto split1 = p.add_instruction(migraphx::op::split{0, {1, 1, 1}, {1, 1}}, x);
+        p.add_instruction(migraphx::op::add{}, split1, split0);
+        return p;
+    }
+};
+
+struct test_split_selector_2 : verify_program<test_split_selector_2>
+{
+    migraphx::program create_program() const
+    {
+        migraphx::program p;
+        migraphx::shape s{migraphx::shape::int32_type, {3, 3, 3}};
+        migraphx::shape s2{migraphx::shape::int32_type, {27}};
+        auto x      = p.add_parameter("x", s);
+        auto split0 = p.add_instruction(migraphx::op::split{1, {1, 1, 1}, {1, 1}}, x);
+        auto split1 = p.add_instruction(migraphx::op::split{1, {1, 1, 1}, {2, 2}}, x);
+        p.add_instruction(migraphx::op::add{}, split1, split0);
+        return p;
+    }
```

Comment:
Can you add a test case for horizontal fusion using the gpu?

Reply:
added.